home/whatsapp.py

```diff
@@ -99,37 +99,131 @@ def from_locale(cls, locale: Locale) -> "WhatsAppLanguage":
+        headers=headers,
+        data=json.dumps(data, indent=4),
+    )
+
+    upload_details = {
+        "upload_session_id": response.json()["id"],
+        "upload_file": image_obj.file,
+    }
+
+    response.raise_for_status()
+    return upload_details
+
+
+def upload_image(image_obj: Image) -> str:
+    """
+    Uploads an image to the Turn API, and returns an image handle reference to use in the template
+    """
+    upload_details = get_upload_session_id(image_obj)
+    url = urljoin(
+        settings.WHATSAPP_API_URL,
+        f"graph/{upload_details['upload_session_id']}",
+    )
+
+    headers = {
+        "file_offset": "0",
+    }
+    files_data = {
+        "file": (upload_details["upload_file"]).open("rb"),
+    }
+    form_data = {
+        "number": settings.FB_BUSINESS_ID,
+        "access_token": settings.WHATSAPP_ACCESS_TOKEN,
+    }
+    response = requests.post(url=url, headers=headers, files=files_data, data=form_data)
+    response.raise_for_status()
+    return response.json()["h"]
+
+
+def create_whatsapp_template_image(image_obj: Image) -> dict[str, Any]:
+    image_handle = upload_image(image_obj)
+
+    return {
+        "type": "HEADER",
+        "format": "IMAGE",
+        "example": {"header_handle": [image_handle]},
+    }
+
+
+def submit_whatsapp_template(
     name: str,
-    body: str,
     category: str,
     locale: Locale,
-    quick_replies: Iterable[str] = (),
-    image_id: int | None = None,
-    example_values: Iterable[str] | None = None,
-) -> None:
-    """
-    Create a WhatsApp template through the WhatsApp Business API.
+    components: list[dict[str, Any]],
+) -> str:
 
-    """
+    url = urljoin(
+        settings.WHATSAPP_API_URL,
+        f"graph/v14.0/{settings.FB_BUSINESS_ID}/message_templates",
+    )
+    headers = {
+        "Authorization": f"Bearer {settings.WHATSAPP_ACCESS_TOKEN}",
+        "Content-Type": "application/json",
+    }
 
-    components = create_whatsapp_template_submission(
-        body, quick_replies, example_values
+    data = {
+        "category": category,
+        "name": name,
+        "language": WhatsAppLanguage.from_locale(locale).value,
+        "components": components,
+    }
+    response = requests.post(
+        url,
+        headers=headers,
+        data=json.dumps(data, indent=4),
     )
-    if image_id is not None:
-        image_obj = Image.objects.get(id=image_id)
-        components.append(create_whatsapp_template_image(image_obj))
+    json_response_content = json.loads(response.content)
 
-    submit_whatsapp_template(name, category, locale, components)
+    result_string = ""
+    # check response code
+    if response.ok:
+        if json_response_content.get("id") is not None:
+            result_string = f"Template Submission OK. Template ID = {json_response_content.get('id')}"
+        return result_string
 
+    result_string = (
+        f"Template Submission ERROR. {response.json()['error']['error_user_msg']} "
+    )
```

Comment:
With this, we have no idea if the result of this function call was successful or not. Maybe this should raise an exception with a custom type that we can catch, with this as the exception message? That way we can determine if the call was successful or not, by whether or not it throws that exception.